### ISSUE TITLE
apiserver along with entire control plane crashes during initialization in previously long running Kubernetes Cluster

### ISSUE THREAD

**Issue by slecrenski** [kubernetes/kubernetes#66477]
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:

We've been running a Kubernetes Cluster for more than a year now.  We have upgraded a few time.  Now all of a sudden we are receiving this and the apiserver/control plane crashes and all docker processes crash on the  master server.

```bash
I0721 22:07:02.680030       1 trace.go:76] Trace[106718694]: "Create /api/v1/namespaces/default/events" (started: 2018-07-21 22:06:58.675783398 +0000 UTC m=+4.884287964) (total time: 4.004207924s):
Trace[106718694]: [4.001371134s] [4.001309883s] About to store object in database
I0721 22:07:04.004239       1 trace.go:76] Trace[1059683869]: "Create /api/v1/nodes" (started: 2018-07-21 22:07:00.000050761 +0000 UTC m=+6.208555347) (total time: 4.004159133s):
Trace[1059683869]: [4.000827818s] [4.000689739s] About to store object in database
I0721 22:07:04.558582       1 trace.go:76] Trace[1586391535]: "Create /api/v1/namespaces/kube-system/events" (started: 2018-07-21 22:07:00.555543641 +0000 UTC m=+6.764048212) (total time: 4.003014383s):
Trace[1586391535]: [4.000529849s] [4.000475957s] About to store object in database
I0721 22:07:06.272917       1 trace.go:76] Trace[259818831]: "Create /api/v1/namespaces/kube-system/events" (started: 2018-07-21 22:07:02.263972787 +0000 UTC m=+8.472477353) (total time: 4.008912365s):
Trace[259818831]: [4.001283782s] [4.001230604s] About to store object in database
I0721 22:07:06.301026       1 trace.go:76] Trace[631370988]: "Create /apis/authentication.k8s.io/v1/tokenreviews" (started: 2018-07-21 22:07:02.298960233 +0000 UTC m=+8.507464797) (total time: 4.002038719s):
Trace[631370988]: [4.000743276s] [4.000632663s] About to store object in database
I0721 22:07:06.684349       1 trace.go:76] Trace[436079227]: "Create /api/v1/namespaces/default/events" (started: 2018-07-21 22:07:02.682089683 +0000 UTC m=+8.890594240) (total time: 4.002234314s):
Trace[436079227]: [4.000627239s] [4.000581775s] About to store object in database
I0721 22:07:46.792024       1 autoregister_controller.go:160] Shutting down autoregister controller
I0721 22:07:46.792566       1 crd_finalizer.go:254] Shutting down CRDFinalizer
I0721 22:07:46.792575       1 apiservice_controller.go:124] Shutting down APIServiceRegistrationController
I0721 22:07:46.792581       1 crdregistration_controller.go:139] Shutting down crd-autoregister controller
I0721 22:07:46.792588       1 available_controller.go:274] Shutting down AvailableConditionController
I0721 22:07:46.792981       1 controller.go:90] Shutting down OpenAPI AggregationController
I0721 22:07:46.793140       1 serve.go:129] Stopped listening on 0.0.0.0:6443
I0721 22:07:46.793167       1 customresource_discovery_controller.go:163] Shutting down DiscoveryController
I0721 22:07:46.793562       1 naming_controller.go:285] Shutting down NamingConditionController
```

```bash
/usr/bin/etcd --listen-client-urls=http://127.0.0.1:2379 --advertise-client-urls=http://127.0.0.1:2379 --data-dir=/kubernetes/etcd-storage/etcd --debug --grpc-keepalive-timeout=0 --proxy='off' --experimental-initial-corrupt-check='true' --election-timeout=50000 --heartbeat-interval 500
```

```bash
2018-07-21 20:01:20.528948 D | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Canceled desc = context canceled")
2018-07-21 20:01:20.528953 D | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Canceled desc = context canceled")
2018-07-21 20:01:20.528989 D | etcdserver/api/v3rpc: failed to receive lease keepalive request from gRPC stream ("rpc error: code = Canceled desc = context canceled")
2018-07-21 20:01:20.528998 D | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Canceled desc = context canceled")
2018-07-21 20:01:20.529005 D | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Canceled desc = context canceled")
```

```bash
etcdctl cluster-health
member 8e9e05c52164694d is healthy: got healthy result from http://127.0.0.1:2379
cluster is healthy
```

**What you expected to happen**:  I would expect the control plane to never crash.

**How to reproduce it (as minimally and precisely as possible)**:
This one will be difficult for you to reproduce.  But I am unable to start any of the kubernetes control plan.  The entire control plane crashes after the apiserver crashes.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.9.6
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.6", GitCommit:"9f8ebd171479bec0ada837d7ee641dec2f8c6dd1", GitTreeState:"clean", BuildDate:"2018-03-21T15:21:50Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Unable to connect to the server: Tunnel Connection Failed
- Cloud provider or hardware configuration: On-Premise
- OS (e.g. from /etc/os-release): Red Hat Enterprise Linux Server release 7.5 (Maipo)
- Kernel (e.g. `uname -a`): 3.10.0-862.6.3.el7.x86_64 #1 SMP Fri Jun 15 17:57:37 EDT 2018 x86_64 x86_64 x86_64 GNU/Linux
- Install tools: yum
- Others:

**Comment by hzxuzhonghu** [comment]
I can not see what caused your apiserver to exit. But I can say that the `goroutine 24774 [running]:` is not an error. It indicates a new request is coming when apiserver is shutting down, We have to deny this request for graceful shut down.

**Comment by hzxuzhonghu** [comment]
`I0721 22:07:06.272917       1 trace.go:76] Trace[259818831]: "Create /api/v1/namespaces/kube-system/events" (started: 2018-07-21 22:07:02.263972787 +0000 UTC m=+8.472477353) (total time: 4.008912365s):
Trace[259818831]: [4.001283782s] [4.001230604s] About to store object in database`

Do you have any webhook audit or admission controller?

**Comment by slecrenski** [comment]
This was a strange one for me as I've never seen it before and it worries me a bit..  I rebooted the compute node that was providing this error message.

```bash
Message from syslogd@nodehostname at Jul 22 06:55:52 ...
 kernel:unregister_netdevice: waiting for eth0 to become free. Usage count = 1
```

After rebooting the node responding with the error message above I restarted the kubelet on the master node and now the control plane appears to be staying online.  Perhaps there is some error condition that needs to be reviewed to prevent the entire control plane from faulting.  Or perhaps there is some other underlying issue with my network/storage but I spent a few hours reviewing the logs of all of the control plane components and it seemed to only crash all of the control plane components after the apiserver failed.  Sorry for the lack of detail as I know it would be hard to make anything useful out of this.
